File: enode/src/main/java/org/enodeframework/messaging/Message.java
package org.enodeframework.messaging;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;

public interface Message {
    /**
     * Represents the unique identifier of the message.
     */
    String getId();

    void setId(String id);

    /**
     * Represents the timestamp of the message.
     */
    Date getTimestamp();

    void setTimestamp(Date timestamp);

    /**
     * Represents the extension key/values data of the message.
     */
    Map<String, Object> getItems();


    void setItems(Map<String, Object> items);

    /**
     * Merge the givens key/values into the current Items.
     */
    default void mergeItems(Map<String, Object> data) {
        if (data == null || data.isEmpty()) {
            return;
        }
        if (this.getItems() == null) {
            this.setItems(new HashMap<>(data));
            return;
        }
        this.getItems().putAll(data);
    }
}

File: enode/src/main/java/org/enodeframework/messaging/Message.kt
package org.enodeframework.messaging

import java.util.*

interface Message {
    /**
     * Represents the unique identifier of the message.
     */
    var id: String

    /**
     * Represents the timestamp of the message.
     */
    var timestamp: Date

    /**
     * Represents the extension key/values data of the message.
     */
    var items: MutableMap<String, Any>

    /**
     * Merge the givens key/values into the current Items.
     */
    fun mergeItems(data: MutableMap<String, Any>) {
        if (data.isEmpty()) {
            return
        }
        items.putAll(data)
    }
}
